src/NodeApi/Runtime/NodeJSRuntime.cs

```diff
@@ -0,0 +1,2277 @@
+        napi_get_node_version;
+    private delegate* unmanaged[Cdecl]<napi_threadsafe_function, nint, napi_status>
+        napi_get_threadsafe_function_context;
+    private delegate* unmanaged[Cdecl]<napi_env, nint, napi_status>
+        napi_get_uv_event_loop;
+    private delegate* unmanaged[Cdecl]<napi_env, napi_value, nint, napi_status>
+        napi_is_buffer;
+    private delegate* unmanaged[Cdecl]<
+        napi_env,
+        napi_async_context,
+        napi_value,
+        napi_value,
+        nuint,
+        nint,
+        nint,
+        napi_status>
+        napi_make_callback;
+    private delegate* unmanaged[Cdecl]<nint, void>
+        napi_module_register;
+    private delegate* unmanaged[Cdecl]<
+        napi_env, napi_value, napi_async_context, nint, napi_status>
+        napi_open_callback_scope;
+    private delegate* unmanaged[Cdecl]<napi_env, napi_async_work, napi_status>
+        napi_queue_async_work;
+    private delegate* unmanaged[Cdecl]<napi_env, napi_threadsafe_function, napi_status>
+        napi_ref_threadsafe_function;
+    private delegate* unmanaged[Cdecl]<
+        napi_threadsafe_function, napi_threadsafe_function_release_mode, napi_status>
+        napi_release_threadsafe_function;
+    private delegate* unmanaged[Cdecl]<napi_async_cleanup_hook_handle, napi_status>
+        napi_remove_async_cleanup_hook;
+    private delegate* unmanaged[Cdecl]<napi_env, napi_cleanup_hook, nint, napi_status>
+        napi_remove_env_cleanup_hook;
+    private delegate* unmanaged[Cdecl]<napi_env, napi_threadsafe_function, napi_status>
+        napi_unref_threadsafe_function;
+    private delegate* unmanaged[Cdecl]<napi_env, nint, napi_status>
+        node_api_get_module_file_name;
+
+    //--------------------------------------------------------------------------------------
+    // Embedding APIs
+    //--------------------------------------------------------------------------------------
+
+    private delegate* unmanaged[Cdecl]<napi_env, napi_value, nint, napi_status>
+        napi_await_promise;
+    private delegate* unmanaged[Cdecl]<
+        napi_platform, napi_error_message_handler, nint, nint, napi_status>
+        napi_create_environment;
+    private delegate* unmanaged[Cdecl]<
+        int, nint, int, nint, napi_error_message_handler, nint, napi_status>
+        napi_create_platform;
+    private delegate* unmanaged[Cdecl]<napi_env, nint, napi_status>
+        napi_destroy_environment;
+    private delegate* unmanaged[Cdecl]<napi_platform, napi_status>
+        napi_destroy_platform;
+    private delegate* unmanaged[Cdecl]<napi_env, napi_status>
+        napi_run_environment;
+
+#pragma warning restore CS0169
+#pragma warning restore IDE0044
+
+    private delegate* unmanaged[Cdecl]<napi_env, nint, napi_status> napi_get_version;
+
+    public override napi_status GetVersion(napi_env env, out uint result)
+    {
+        if (napi_get_version == null)
```

Comment:
Checks like this are unnecessary. Our current code has a better implementation based on stubs and replacing them with real function pointers.

Reply:
Yes, but that required a lot of extra code to make that work, which adds to maintenance overhead. The cost of a simple null check is negligible (as confirmed by benchmarks), so I prefer the simpler code.